juju/juju
# Acme wtf removal

## Description of change

The worker/uniter tests were compiling an actual jujud from source and using that in full stack tests. Many of the tests were for functionality tested elsewhere, eg does a hook command reject bad input, or does the deployer unpack a charm correctly. And some tests were for things that should have been covered elsewhere eg there were tests which ran an action and tested results but the test for the results watcher should have been in state package where the watcher is implemented.

This PR creates mocks for 2 key components used by the uniter worker
- the runner which handles execution of hooks, commands, actions
- the deployer which unpacks a charm

The worker config has been changed to allow either the real implementation of a mock to be passed in to the uniter.

In addition, it seems that the tests should have been failing - and they do now without additional changes. New hooks were added - relation-created and remove. The relation related tests were not updated to expect these hooks. With changes in this PR, the tests failed until these expectations were updated.

Finally, the obsoleted GitSuite was removed.

## QA steps

deploy both a k8s and vm charm
use show-status-log to check hook execution
use "juju exec --unit foo/0 -- hostname" to check running scripts
use "juju exec --unit foo/0 -- config-get to check running hook commands
run an action also and check the results
